### PR TITLE
EZP-27938: Add tests and perform cleanup in RequestListener

### PR DESCRIPTION
**Pull request by mikadamczyk**
> https://jira.ez.no/browse/EZP-27938

### CODE REVIEW

File: bootstrap.php [hunk 1]
```diff
@@ -0,0 +1 @@
+<?php require_once __DIR__ . '/vendor/autoload.php';
```
Comment (andrerom) [inline bootstrap.php]: can be placed in test folder, example: https://github.com/ezsystems/repository-forms/blob/master/tests/bootstrap.php

File: src/lib/EventListener/RequestListener.php [hunk 2]
```diff
@@ -16,7 +15,12 @@
 
 class RequestListener implements EventSubscriberInterface
 {
-    use ContainerAwareTrait;
+    private $groupsBySiteAccess;
+
+    public function __construct($groupsBySiteAccess)
+    {
+        $this->groupsBySiteAccess = $groupsBySiteAccess;
```
Comment (andrerom) [inline src/lib/EventListener/RequestListener.php]: note that siteaccess can change. If we don't take from container, ideally class should be adapted to expose setter to refresh setting on siteaccess changes, there are some examples around but I'm not the export on this.

Comment (bdunogier) [reply src/lib/EventListener/RequestListener.php]: I'm not sure that is relevant in this context: what gets injected is the list of groups per siteaccess, not the _current_ siteaccess.

The current siteaccess may change dynamically, but the list of siteaccess/siteaccess groups may not, right ?

File: bootstrap.php (hunk 1 above)
Comment (mikadamczyk) [reply bootstrap.php]: yes, but possibly we can have tests also in /src/bundle directory, then the file would be duplicated

File: src/lib/EventListener/RequestListener.php (hunk 2 above)
Comment (mikadamczyk) [reply src/lib/EventListener/RequestListener.php]: as I know, current SiteAccess can be changed, but here we fetch only list all of available groups by SiteAccess. Can it be changed during master request?

Comment (andrerom) [reply src/lib/EventListener/RequestListener.php]: you are right, if it is the global map then it is not subject to issues on this.

Comment (bdunogier) [reply src/lib/EventListener/RequestListener.php]: Based on the name, it is the list of groups for each SA.

Comment (mikadamczyk) [reply src/lib/EventListener/RequestListener.php]: I think it is a global map, not  groups of the current SiteAccess

**Comment by Nattfarinn** [comment]
Please rebase with master (conflicts). :)

File: src/lib/Tests/EventListener/RequestListenerTest.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    private $request;
+
+    /**
+     * @var GetResponseEvent
+     */
+    private $event;
+
+    /**
+     * @var HttpKernelInterface|\PHPUnit_Framework_MockObject_MockObject
```
Comment (webhdx) [inline src/lib/Tests/EventListener/RequestListenerTest.php]: CS, `/** @var (...) */` one liner as above.

File: src/lib/Tests/EventListener/RequestListenerTest.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            $this->request,
+            HttpKernelInterface::SUB_REQUEST
+        );
+        $this->assertNull($this->requestListener->onKernelRequest($this->event));
+    }
+
+    public function testOnKernelRequestAllowAccessWithoutSiteAccess()
+    {
+        $this->request->attributes->set('siteaccess', 'not_siteaccess_object');
+        $this->assertNull($this->requestListener->onKernelRequest($this->event));
```
Comment (webhdx) [inline src/lib/Tests/EventListener/RequestListenerTest.php]: Empty line. Also check other occurrences below.

**Comment by mikadamczyk** [comment]
Hi @webhdx, I updated PR.

File: src/lib/EventListener/RequestListener.php [hunk 5]
```diff
@@ -16,7 +15,12 @@
 
 class RequestListener implements EventSubscriberInterface
 {
-    use ContainerAwareTrait;
+    private $groupsBySiteAccess;
```
Comment (webhdx) [inline src/lib/EventListener/RequestListener.php]: As far as I understand this is an array, please enclose type hinting information.

File: src/lib/EventListener/RequestListener.php [hunk 6]
```diff
@@ -16,7 +15,12 @@
...
+
+    public function __construct($groupsBySiteAccess)
```
Comment (webhdx) [inline src/lib/EventListener/RequestListener.php]: Type hinting + docblock

File: src/lib/Tests/EventListener/RequestListenerTest.php [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    /** @var RequestListener */
+    private $requestListener;
+
+    /** @var Request */
+    private $request;
+
+    /** @var GetResponseEvent */
+    private $event;
+
+    /** @var HttpKernelInterface|\PHPUnit_Framework_MockObject_MockObject */
```
Comment (webhdx) [inline src/lib/Tests/EventListener/RequestListenerTest.php]: New PHPUnit has namespaces support which we should use.

Comment (mikadamczyk) [reply src/lib/Tests/EventListener/RequestListenerTest.php]: As I see, createMock() return PHPUnit_Framework_MockObject_MockObject

**Pull request merged**